$OpenBSD$

Debian's patch modified:
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=689642

Fix errors "conflicting declaration 'typedef long int int64'" and "conflicting
declaration 'typedef long unsigned int uint64'" on amd64.

--- src/nvcore/DefsGnucLinux.h.orig	Fri May 14 19:59:28 2010
+++ src/nvcore/DefsGnucLinux.h	Mon Oct 22 01:04:34 2012
@@ -2,6 +2,8 @@
 #error "Do not include this file directly."
 #endif
 
+#include <limits.h>
+
 // Function linkage
 #define DLL_IMPORT
 #if __GNUC__ >= 4
@@ -58,8 +60,13 @@ typedef signed short        int16;
 typedef unsigned int        uint32;
 typedef signed int          int32;
 
+#if LONG_BIT == 32
 typedef unsigned long long  uint64;
 typedef signed long long    int64;
+#else
+typedef unsigned long       uint64;
+typedef signed long         int64;
+#endif
 
 // Aliases
 typedef uint32              uint;
